### DISCUSSION TITLE
LTSP clients unable to run Mariadb

### DISCUSSION THREAD

**Discussion by sudhanshubhasin** [ltsp/ltsp#855] [General]
Here is the error from the LTSP Client

root@ltsp84:/home/users01# systemctl start mariadb
Job for mariadb.service failed because a fatal signal was delivered causing the control process to dump core.
See "systemctl status mariadb.service" and "journalctl -xe" for details.

There is no error on LTSP Server - it is running and active

**Comment by alkisg** [comment]
Run the commands that it proposes, so that you see the error logs.
Which database are you running?
Are you sure you want a copy of that database to be in the read-only image, and available publicly to the whole network?

**Reply by sudhanshubhasin** [reply]
Which database are you running ?

**LTSP Server: mariadb.service - MariaDB 10.6.14 database server
on Server: is running** 

on LTSP 
ystemctl start mariadb
Job for mariadb.service failed because a fatal signal was delivered causing the control process to dump core.
See "systemctl status mariadb.service" and "journalctl -xe" for details.


Are you sure you want a copy of that database to be in the read-only image, and available publicly to the whole network?


We want the LTSP client to accept data and send to LTSP Server running Mariadb

**Comment by alkisg** [comment]
You need to run the commands that it mentions to see the logs:
`systemctl status mariadb.service` and `journalctl -xe`.

> Which database are you running ?

I meant which software are you running that uses MariaDB for storage. E.g. are you running a web server with a php/mysql application? And which application is that, what is its name? See some applications there: https://mariadb.com/kb/en/applications-supporting-mariadb/
Such web applications usually should only run on the server, not on the clients. The clients then connect to the server via http(s), i.e. a browser.

Finally, note that the following directories are automatically excluded when you run `ltsp image`, exactly for the reasons mentioned above:

```
$ grep sql image.excludes
etc/mysql/debian.cnf
var/lib/mysql/*
```

**Comment by sudhanshubhasin** [comment]
[https://tr.cloudmagic.com/h/v6/emailtag/tag/2.0/1691680658/e9b135c8d2df62cd3f0a2459edd3af78/38/5de8f39ffe08c8666659294f36f4f319/9b5d7dd31573a9202c63b5a583ef565e/9efab2399c7c560b34de477b9aa0a465/newton.gif]

We are using a bash script to accept barcode reader to feed data into inventory and server is running fine with updating data in mariadb but it is only client having issues sending data to mariadb


via Newton Mail<https://cloudmagic.com/k/d/mailapp?ct=pa&cv=11.0.0&pv=13&source=email_footer_2>
On Thu, Aug 10, 2023 at 8:20pm, Alkis Georgopoulos ***@***.******@***.***>> wrote:


You need to run the commands that it mentions to see the logs:
systemctl status mariadb.service and journalctl -xe.

Which database are you running ?

I meant which software are you running that uses MariaDB for storage. E.g. are you running a web server with a php/mysql application? And which application is that, what is its name? See some applications there: https://mariadb.com/kb/en/applications-supporting-mariadb/<https://tr.cloudmagic.com/h/v6/link-track/1.0/1691680676326386-69b66b59-3e90-da58-d3b3-40ec56095e39/1691680658/e9b135c8d2df62cd3f0a2459edd3af78/9b5d7dd31573a9202c63b5a583ef565e/9efab2399c7c560b34de477b9aa0a465?redirect_uri=https://mariadb.com/kb/en/applications-supporting-mariadb/>
Such web applications usually should only run on the server, not on the clients. The clients then connect to the server via http(s), i.e. a browser.

Finally, note that the following directories are automatically excluded when you run ltsp image, exactly for the reasons mentioned above:

$ grep sql image.excludes
etc/mysql/debian.cnf
var/lib/mysql/*

**Comment by alkisg** [comment]
Then you need to convert your bash script into a web application, because mariadb isn't a network database, it can't maintain a single database while running in multiple workstations.

This isn't an LTSP related issue.
Imagine that you have 10 standalone workstations without LTSP.
How will your bash script work there?
Once you answer that question, then it'll be easy to do the same thing with LTSP.

**Comment by sudhanshubhasin** [comment]
[https://tr.cloudmagic.com/h/v6/emailtag/tag/2.0/1691681415/e9b135c8d2df62cd3f0a2459edd3af78/38/5de8f39ffe08c8666659294f36f4f319/9fe8f5f722da82d246c03e34e3a76383/9efab2399c7c560b34de477b9aa0a465/newton.gif]

Ltsp server running bash script and using reader to enter data working great .


Ltsp client is not able to start mariadb services


via Newton Mail<https://cloudmagic.com/k/d/mailapp?ct=pa&cv=11.0.0&pv=13&source=email_footer_2>
On Thu, Aug 10, 2023 at 8:54pm, Alkis Georgopoulos ***@***.******@***.***>> wrote:


Then you need to convert your bash script into a web application, because mariadb isn't a network database, it can't maintain a single database while running in multiple workstations.

This isn't an LTSP related issue.
Imagine that you have 10 standalone workstations without LTSP.
How will your bash script work there?
Once you answer that question, then it'll be easy to do the same thing with LTSP.

**Comment by sudhanshubhasin** [comment]
Closing as find alternate method

**Comment by sudhanshubhasin** [comment]
closed
